### ISSUE TITLE
Play midi file: differentiate between Tracks

### ISSUE THREAD

**Issue by EmanuelFeru** [mido/mido#287]
Hi,

First of all thank you very much for this very nice library, it saves so much time and effort in working with MIDI files. I am trying to implement Learning without Synthesia in this project (see [this item](https://github.com/onlaj/Piano-LED-Visualizer/issues/180)). And for that I need to be able to play the MIDI file step wise and separate per track. 

I have a question, or maybe you have suggestions on how to do it. In the example from documentation:
```python
for msg in MidiFile('song.mid'):
    time.sleep(msg.time)
    if not msg.is_meta:
        port.send(msg)
```
 is there a way to check to which Track a `msg` is belonging? Something like:
 ```python
 if msg.track == 'Track 0':
     # ....
 elif msg.track == 'Track 1':
     # ....
 ```
 I want to light up the LEDs to a different color per Tracks (Track 0: right hand, Track 1: left hand). So I need to be able to play one Track only or both Tracks in the same time with the ability to differentiate between tracks.

**Comment by EmanuelFeru** [comment]
I have just seen this Discussion page https://github.com/mido/mido/discussions/257
We can discuss there and close this topic if you find it more appropriate.

**Comment by EmanuelFeru** [comment]
I did some digging. I found that the function below is used to merge tracks in one track. The only draw back with this is that once the tracks are merged the messages are not knows to which Track they are belonging.
https://github.com/mido/mido/blob/2f53ba0bfa4c0fc5949bd5802d9cb03366e34317/mido/midifiles/tracks.py#L102-L114

**Comment by EmanuelFeru** [comment]
I think I have an initial maybe "quick and dirty" solution. I store the `track` information in the `channel` field so the information is not lost after merging, i.e.:
```python
# Assign Tracks to different channels before merging to know the message origin
for k in range(len(mid.tracks)):
    for msg in mid.tracks[k]:
        if not msg.is_meta:
            msg.channel = k
```
Then the full code becomes as below. Which gives me the desired behavior. I can select which channel(s) to play with an if statement.
```python

import mido
import time

def get_tempo(mid):
    for msg in mid:     # Search for tempo
        if msg.type == 'set_tempo':
            return msg.tempo
    return 500000       # If not found return default tempo

# Open port
outport = mido.open_output()
print(outport)

mid = mido.MidiFile("my_piano_song.mid")

# Get tempo
tempo =  get_tempo(mid)

# Assign Tracks to different channels before merging to know the message origin
for k in range(len(mid.tracks)):
    for msg in mid.tracks[k]:
        if not msg.is_meta:
            msg.channel = k

# Merge tracks and play
all_tracks = mido.merge_tracks(mid.tracks)
for msg in all_tracks:
    time.sleep(mido.tick2second(msg.time, mid.ticks_per_beat, tempo))
    if not msg.is_meta:
        if (msg.channel == 0):  # Play Track 0
            outport.send(msg)
        if (msg.channel == 1):  # Play Track 1
            outport.send(msg)
```

**Comment by pr0jectile** [comment]
Awesome work here @EmanuelFeru! Just came across this.

Just a quick one, you might want to be using `if (msg.channel == 1):` and `if (msg.channel == 2):` since track 0 is for meta messages like `set_tempo`, `time_signature`, `key_signature` etc.

I'm trying to modify your version of the full code to work with Blender 2.9 piano animations but finding it tricky with making it work properly with framerates – new to animation scripting here.

Let me know if you can help :)

**Comment by EmanuelFeru** [comment]
Thank you for the input @pr0jectile! I will change to your suggestion. It is the first time I am working with MidiFiles so I don't know the convention exactly. However, for some reason in my MidiFile Track 0 contains notes as well (The Right hand) while Track 1 is the Left hand.

---

> I'm trying to modify your version of the full code to work with Blender 2.9 piano animations but finding it tricky with making it work properly with framerates – new to animation scripting here.

I need to understand a bit in more detail to be able to help. However, If I understand correctly you need to have a synchronization between the frame rate of the animation and the notes time delay. Then, I think you just need to multiply the time delay in `seconds` with the frame rate `fps` in `frames/second`. So, the line with the time delay becomes:
```python
time.sleep(fps * mido.tick2second(msg.time, mid.ticks_per_beat, tempo))
```
So for example if the time delay is `1 s` and the `fps = 25` then the delay will be `1 * 25 = 25 frames`. Could be that some tricks will be needed because of rounding errors to stay in sync. Because if `delay = 0.1 s`, then you have `2.5 frames` but the frame rate is an integer. That needs to be checked.

**Comment by pr0jectile** [comment]
No problem @EmanuelFeru, I'm sure someone else will be able to confirm the following too but generally I've found that the MIDI files I've created using something like Sibelius use Track 0 for information, and Track 1 to Track 15 is actual MIDI notes. 

The way I create my left / right hand data is by splitting the bass notes to a separate track and the treble notes are left on another track.

Might also depend on the MIDI Type which I've not played around with myself but my MIDI tracks are Type 1.

I've added the following below `mid = mido.MidiFile("my_piano_song.mid")` to get some more details about the MIDI file:

```python
print("TYPE: " + str(mid.type))
print("LENGTH: " + str(mid.length))
print("TICKS PER BEAT: " + str(mid.ticks_per_beat))
```

Maybe this can clarify in your case with the MIDI file you're using.

Thanks for the input regarding fps, I'll definitely have to play around with this.

I'm trying to get the MIDI note on/off times converted to fps of the animation scene.

For example, if a note being played at `2m 4s` for a duration of `10 seconds` taking `set_tempo` into account, in a `60 fps` animation, at `2m 4s` **on_animation** should trigger then at `2m 14s` **off_animation** should trigger.
